engine.js

```diff
@@ -931,6 +931,13 @@ var accumulateResult = function(ifResult, thenResult) {
     };
 
     HMDAEngine.isValidCensusTractCombo = function(censusTract, metroArea, fipsState, fipsCounty) {
+        if (censusTract === 'NA' && metroArea === 'NA' && fipsState === 'NA' && fipsCounty === 'NA') {
+            return Promise.resolve()
```

Comment:
Don't actually need to create a new Promise() here, can just `return true;`. I get that you were probably going for consistency in return type, but in this case, it's unnecessary.  Test can also be updated to just check for direct return instead of promise.